cmd/agent/app/httpserver/server.go

```diff
@@ -0,0 +1,184 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package httpserver
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/uber/jaeger-lib/metrics"
+
+	tSampling "github.com/uber/jaeger/thrift-gen/sampling"
+)
+
+const mimeTypeApplicationJSON = "application/json"
+
+var (
+	errBadRequest = errors.New("Bad request")
+)
+
+// NewHTTPServer creates a new server that hosts an HTTP/JSON endpoint for clients
+// to query for sampling strategies and baggage restrictions.
+func NewHTTPServer(hostPort string, manager Manager, mFactory metrics.Factory) *http.Server {
+	handler := newHTTPHandler(manager, mFactory)
+	mux := http.NewServeMux()
+	mux.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
+		handler.serveSamplingHTTP(w, r, true /* thriftEnums092 */)
+	})
+	mux.HandleFunc("/sampling", func(w http.ResponseWriter, r *http.Request) {
+		handler.serveSamplingHTTP(w, r, false /* thriftEnums092 */)
+	})
+	mux.HandleFunc("/baggage", func(w http.ResponseWriter, r *http.Request) {
+		handler.serveBaggageHTTP(w, r)
+	})
+	return &http.Server{Addr: hostPort, Handler: mux}
+}
+
+func newHTTPHandler(manager Manager, mFactory metrics.Factory) *httpHandler {
+	handler := &httpHandler{manager: manager}
+	metrics.Init(&handler.metrics, mFactory, nil)
+	return handler
+}
+
+type httpHandler struct {
+	manager Manager
+	metrics struct {
```

Comment:
We should really just replace these with rpc-metrics from jaeger-client (we can run the tracer with sampling off). Are there any metrics that are not captured by rpc-metrics?